lib/httpClient.js

```diff
@@ -0,0 +1,202 @@
+
+class HttpClient {
+  constructor(baseUrl, requestHeadersToProxy, baseOptions = {}, logger) {
+    this.baseUrl = baseUrl
+    this.requestHeadersToProxy = requestHeadersToProxy
+    this.baseOptions = baseOptions
+    this.logger = logger || Pino({ level: 'silent' })
+
+    const httpsAgent = getHttpsAgent(baseOptions)
+    this.axios = axios.create({
+      ...httpsAgent ? { httpsAgent } : {},
+      timeout: baseOptions.timeout,
+    })
+  }
+
+  async get(path, options = {}) {
+    return this.makeCall(path, undefined, {
+      ...options,
+      method: 'GET',
+    })
+  }
+
+  async post(path, payload, options = {}) {
+    return this.makeCall(path, payload, {
+      ...options,
+      method: 'POST',
+    })
+  }
+
+  async put(path, payload, options = {}) {
+    return this.makeCall(path, payload, {
+      ...options,
+      method: 'PUT',
+    })
+  }
+
+  async patch(path, payload, options = {}) {
+    return this.makeCall(path, payload, {
+      ...options,
+      method: 'PATCH',
+    })
+  }
+
+  async delete(path, payload, options = {}) {
+    return this.makeCall(path, payload, {
+      ...options,
+      method: 'DELETE',
+    })
+  }
+
+  async makeCall(path, payload, options) {
+    let headers = getHeaders(options, this.requestHeadersToProxy, this.baseOptions)
+    if (typeof payload === 'object') {
+      headers = {
+        'content-type': 'application/json;charset=utf-8',
+        ...headers,
+      }
+    }
+    if (payload === undefined) {
+      headers = {
+        'content-length': '0',
+        'content-type': 'text/html',
+        ...headers,
+      }
+    }
+    const url = getUrl(this.baseUrl, path, this.baseOptions)
+    const httpsAgent = getHttpsAgent(options)
+    const errorMessageKey = getErrorMessageKey(this.baseOptions, options)
+    try {
+      const validateStatus = getValidateStatus(options)
+      const response = await this.axios({
+        url,
+        method: options.method,
+        headers,
+        data: getData(payload),
+        responseType: getResponseType(options),
+        params: options.query,
+        ...validateStatus ? { validateStatus } : {},
+        timeout: options.timeout,
+        proxy: options.proxy,
+        ...httpsAgent ? { httpsAgent } : {},
+      })
+      return {
+        statusCode: response.status,
+        headers: response.headers,
+        payload: response.data,
+      }
+    } catch (error) {
+      if (error.response) {
+        const errorMessage = getErrorMessage(error.response, options.returnAs, errorMessageKey)
+        const responseError = new Error(errorMessage)
+        responseError.headers = error.response.headers
+        responseError.statusCode = error.response.status
+        responseError.payload = error.response.data
+        throw responseError
+      }
+      const errToReturn = new Error(error.message)
+      errToReturn.code = error.code
+      throw errToReturn
+    }
+  }
+}
+
+function getErrorMessageKey(baseOptions, options) {
```

Comment:
If options is the first object that you check, why not use it as first parameter of the function?